### PR TITLE
[test] Intrinsics with colon no longer type-checked

### PR DESCRIPTION
**Pull request by eps1lon**
[types/fbt is currently broken in 5.1.0-dev.20230415
](https://www.typescriptlang.org/play?ts=5.1.0-dev.20230415#code/JYWwDg9gTgLgBAKjgQwM5wEoFNkGN4BmUEIcA5FDvmQNwBQdAJlrgDbKVwDmrEARslZwA3nTji4AO2QgsqMHixwAUgGUAGiLESdwSTCxQCiuAEl9UPamC4Aoqyyz96UTrcSyBPjABcWSQCuIGQ+InAA1pIQAO6SAArEYKGoMJaSXHAAvvTu4pnaeXT5dAA8Xr7+QRFRsQkQYAC8AER4uFhgBoxNBXAA9L1wAAIwqAC0WAAeYCww41DEUD0BkpEx8YnNlABWM1hdfQB89EA)

Broke in https://github.com/microsoft/TypeScript/pull/47356

I'm not sure yet whether this was caused by https://github.com/microsoft/TypeScript/pull/51328 or https://github.com/microsoft/TypeScript/pull/53284.

Considering https://github.com/microsoft/TypeScript/pull/53284 because the hovering with the cursor treats the parts before and after the colon as separate types:
5.0:

Screenshot from 2023-04-15 10-26-01

5.1:

Screenshot from 2023-04-15 10-27-06
Screenshot from 2023-04-15 10-27-02

### CODE REVIEW

File: tests/baselines/reference/jsxElementType.errors.txt [hunk 1]
```diff
@@ -197,4 +197,15 @@ tests/cases/compiler/jsxElementType.tsx(99,2): error TS2304: Cannot find name 'U
     <Unresolved foo="abc" />;
      ~~~~~~~~~~
 !!! error TS2304: Cannot find name 'Unresolved'.
+    
+    // regression test for intrinsic containing `:`
+    declare global {
+      namespace JSX {
+        interface IntrinsicElements {
+          ['fbt:enum']: { knownProp: string };
```
Comment (eps1lon) [inline tests/baselines/reference/jsxElementType.errors.txt:205]: Maybe this needs to be declared another way now that https://github.com/microsoft/TypeScript/pull/47356/ has landed? But even if we drop support for that, the error message currently produced is not helpful. It silently treats `<fbt:enum />` as any.

**Comment by DanielRosenwasser** [comment]
@typescript-bot cherry-pick this to revert-47356-feat/7411

**Comment by a-tarasyuk** [comment]
@eps1lon Thank you for taking the time to immediately test the latest changes and identify the new important edge case. I've created a new pull request https://github.com/microsoft/TypeScript/pull/53799 that fixes the bug.

@DanielRosenwasser I noticed that you opened a PR to revert the changes. Would you like me to create a new PR after merging the revert into the main branch?

**Comment by DanielRosenwasser** [comment]
Yeah I just wanted to see if the behavior was changed. If we can get in your fix (just apply my test suggestions, update baselines, and take the PR out of draft state) then I think I can bring it into the TS 5.1 beta.

**Comment by eps1lon** [comment]
Fixed in https://github.com/microsoft/TypeScript/pull/53799

**Pull request closed without merging**